docs/specs/idp-account-linking.md

```diff
@@ -12,47 +12,60 @@ Here is an example of the configuration:
 
 ```yaml
 identity:
-  providers:
-    - alias: google
-      client_id: exampleclientid
-      type: google
-      account_linking:
-        enabled: true
-        idp_claim_key: "email"
-        match_against_claim_key: "email"
+  on_conflict:
+    account_linking:
+      oauth: "merge"
+  oauth:
+    providers:
+      - alias: google
+        client_id: exampleclientid
+        type: google
+        account_linking:
+          enabled: false
+      - alias: github
+        client_id: exampleclientid
+        type: github
+        account_linking:
+          enabled: true
+          match_by_claim: "email"
 ```
 
-- The `account_linking` object was added.
+- The `account_linking` object was added to provider config.
 - `account_linking.enabled`: A boolean indicates if account linking is enabled for this provider.
-- `account_linking.idp_claim_key`: The key which used to read a claim value from from userinfo of the idp.
-- `account_linking.match_against_claim_key`: The key which used to read a claim value from from userinfo of authgear. If the value matched with the value read using `account_linking.idp_claim_key` in the idp userinfo, an account linking will be performed.
+- `account_linking.match_by_claim`: The claim which used to match the oauth account to an authgear user. For example `"email"`. It could be a standard claim, or a custom attribute. Sensible options should be displayed in portal, but no valiation will be done at server because we would like to keep the flexibility to use any claim for any provider.
+
+- The `identity.on_conflict.account_linking.oauth` config was added to specify the behavior if an oauth account linking was triggered. The possible values are:
+  - `"error"`: Reject the signup with an error. This is the default.
+  - `"merge"`: Merge the new oauth identity to the existing user after an authentication process.
 
 ## Trigger conditions and actions
 
 - Whenever an oauth identity is used, and there is no existing user for that identity.
 
   1. If `account_linking.enabled` is not `true`, do nothing. Else,
```

Comment:
Step 1 is not backward compatible. It is because account_link.enabled is false by default. The backward compatible behavior should be `"error"`.

Reply:
For backward compatibility, modify the default value as mentioned in https://github.com/authgear/authgear-server/pull/4096#discussion_r1555413513